uniflow/op/extract/load/utils.py

```diff
@@ -0,0 +1,63 @@
+"""Defines a function to read files from Amazon S3, URLs, or local paths in text or binary mode."""
+
+import requests
+
+
+def read_file(source: str, mode: str = "r"):
+    """
+    Reads a file from an S3 bucket, a URL, or the local filesystem based on the source path provided.
+
+    Parameters:
+    - source (str): The path to the file. This can be an S3 bucket path (s3://bucket_name/file_path),
+      a URL (http:// or https://), or a local file path.
+    - mode (str): The mode in which to open the file. Use 'r' for text mode and 'rb' for binary mode.
+      Default is 'r'.
```

Comment:
nit: Args because we we using [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).